### PR TITLE
fix(init) make tests green and pass correct options to luasec_opts

### PR DESCRIPTION
**Pull request by ghost**
ngx.ssl is only available in openresty land, whereas this plugin's CI does not have it enabled/installed to run tests.

Also fix a regression where luasec_opts had incorrect parameters.

### CODE REVIEW

File: pgmoon/init.lua [hunk 1]
```diff
@@ -6,8 +6,12 @@ do
   local _obj_0 = require("pgmoon.bit")
   rshift, lshift, band = _obj_0.rshift, _obj_0.lshift, _obj_0.band
 end
-local pl_file = require("pl.file")
-local ssl = require("ngx.ssl")
+local pl_file
+local ssl
+if ngx then
```
Comment (ghost) [inline pgmoon/init.lua:11]: `if ngx then ... end` is one way to check if are running in openresty

File: pgmoon/init.moon [hunk 2]
```diff
@@ -2,8 +2,13 @@ socket = require "pgmoon.socket"
...
-ssl = require "ngx.ssl"
+
+local pl_file
+local ssl
+
+if ngx
+  pl_file = require "pl.file"
+  ssl = require "ngx.ssl"
 
 unpack = table.unpack or unpack
```
Comment (ghost) [inline pgmoon/init.moon:14]: is moonscript code necessary for us (kong) since we always run in openresty with luajit?

Comment (dndx) [reply pgmoon/init.moon:14]: I think the Lua code are compiled from `.moon` script, so it is best to not edit the `.lua` file directly, as any changes made may be lost.

File: pgmoon/init.lua (hunk 1 above)
Comment (dndx) [reply pgmoon/init.lua:11]: The usual we we do this is to use `pcall(require, "foo")`, but I'm fine with `if ngx then` as well. No real difference here.

**Review by dndx — APPROVED** [review]

**Pull request merged**